### PR TITLE
updated mission replay sample

### PR DESCRIPTION
**Pull request by ghost**
Updates the mission replay with an MVC architecture

### CODE REVIEW

**Comment by ghost** [comment]
@vGandhi Could you give this a quick review?

File: scene/mission-replay/src/main/java/com/esri/samples/scene/AnimationModel.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+package com.esri.samples.scene;
```
Comment (vGandhi) [inline scene/mission-replay/src/main/java/com/esri/samples/scene/AnimationModel.java:12]: Missing copyright.

File: scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+   * @param mission .csv file name containing the mission data
+   * @return ordered list of mapped key value pairs representing coordinates and rotation parameters for each step of
+   * the mission
+   */
+  private List<Map<String, Object>> getMissionData(String mission) {
+
+    // open a file reader to the mission file that automatically closes after read
+    try (BufferedReader missionFile = new BufferedReader(new InputStreamReader(getClass().getResourceAsStream("/" + mission)))) {
+      List<Map<String, Object>> missionData = new ArrayList<>();
+      missionFile.lines()
```
Comment (vGandhi) [inline scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java:256]: I suggest adding an example of a line that is parsed below.

File: scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+   * @param keyframe index in mission data to show
+   */
+  private void animate(int keyframe) {
+    // get the next position
+    Map<String, Object> datum = missionData.get(keyframe);
+    Point position = (Point) datum.get("position");
+
+    // update the model bean with new parameters
+    planeModel.setAltitude(position.getZ());
+    planeModel.setHeading((double) datum.get("heading"));
```
Comment (vGandhi) [inline scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java]: The string constants can be moved out as class fields. I can be reused in lines 92-94.

Update: Maybe not here, but in lines 275-277.

File: scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+
+  /**
+   * Stops the animation and disposes of application resources.
+   */
+  void terminate() {
+    animation.stop();
+    if (sceneView != null) sceneView.dispose();
+    if (mapView != null) mapView.dispose();
+  }
+}
```
Comment (vGandhi) [inline scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java]: Missing newline.

File: scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+      animation.stop();
+    }
+  }
+
+  /**
+   * Switches the toggle mode on the camera model when the toggle button is clicked.
+   */
+  @FXML
+  private void toggleFollow() {
+    if (followButton.isSelected()) plane2D.getAttributes().put("ANGLE", 0f);
```
Comment (vGandhi) [inline scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java]: In our SDK implementation, we use a verbose "if" like below. As long as all samples follow the same standard, anything works.

```
if() {
 ...
}
```

**Comment by vGandhi** [comment]
@tyle8552 Looks well structured with this refactoring.
Some minor comments. Please add copyright to source files.

**Comment by ghost** [comment]
@vGandhi final check?

**Comment by vGandhi** [comment]
@tyle8552 :+1:

**Pull request merged**